### PR TITLE
Use [u8; 4] for well known cache keys

### LINKED ISSUE #2148: Fixed lenght well_known_cache_keys
Currently cache keys are passed as `Vec<u8>`. `[u8; 4]` or `&'static [u8; 4]` should be used instead.

### PR DESCRIPTION
**Pull request by DarkEld3r**
Closes #2148.

### CODE REVIEW

File: core/client/db/src/cache/mod.rs [hunk 1]
```diff
@@ -253,9 +253,9 @@ impl<'a, Block: BlockT> DbCacheTransaction<'a, Block> {
 pub struct DbCacheSync<Block: BlockT>(pub RwLock<DbCache<Block>>);
 
 impl<Block: BlockT> BlockchainCache<Block> for DbCacheSync<Block> {
-	fn get_at(&self, key: &[u8], at: &BlockId<Block>) -> Option<Vec<u8>> {
+	fn get_at(&self, key: [u8; 4], at: &BlockId<Block>) -> Option<Vec<u8>> {
```
Comment (bkchr) [inline core/client/db/src/cache/mod.rs]: Please revert this change. You can just use it as slice.

File: core/consensus/common/src/lib.rs [hunk 2]
```diff
@@ -120,6 +120,9 @@ impl<T: SyncOracle> SyncOracle for Arc<T> {
 
 /// A list of all well known keys in the cache.
 pub mod well_known_cache_keys {
+	/// The type representing cache keys.
+	pub type Id = [u8; 4];
```
Comment (DarkEld3r) [inline core/consensus/common/src/lib.rs]: @bkchr Perhaps tuple struct should be used instead? I have used `pub type ConsensusEngineId = [u8; 4];` as an example.

Comment (bkchr) [reply core/consensus/common/src/lib.rs]: That is enough, we don't need to use a tuple struct here 👍
